### PR TITLE
Bugfix/37526 custom price line items

### PR DESCRIPTION
**Pull request by LiudmylaMasliuk**

### CODE REVIEW

File: paypal-commercetools-extension/tests/payment.controller.spec.ts [hunk 1]
```diff
@@ -91,6 +61,7 @@ const mockConfigModule = () => {
 mockConfigModule();
 
 import { paymentController } from '../src/controllers/payments.controller';
```
Comment (mbe1987) [inline paypal-commercetools-extension/tests/payment.controller.spec.ts:63]: i guess it should be discountedLineItems

File: paypal-commercetools-extension/tests/payment.controller.spec.ts [hunk 2]
```diff
@@ -256,7 +233,42 @@ async function createValidTransaction() {
...
+const amountPlannedCentsWithTestResult: [number, string, string][] = [
+  [19900, 'same as cart', '199.00'],
+  [200000, 'more than in cart', '2000.00'],
+  [4200, 'less than in cart', '42.00'],
+];
+
 describe('Testing PayPal aftersales', () => {
+  test.each(amountPlannedCentsWithTestResult)(
+    'Create a valid transaction with %s cents, which is %s , leads to processing transaction with %s amount',
+    async (amountPlannedCents, descrption, expectedAmount) => {
```
Comment (mbe1987) [inline paypal-commercetools-extension/tests/payment.controller.spec.ts:245]: typo description

File: paypal-commercetools-extension/src/utils/map.utils.ts [hunk 3]
```diff
@@ -174,6 +175,26 @@ export const mapCommercetoolsLineItemsToPayPalItems = (
...
+export const mapValidCommercetoolsLineItemsToPayPalItems = (
+  matchingAmounts: boolean,
+  isShipped: boolean,
+  lineItems?: LineItem[],
+  locale?: string
+) => {
+  if (!matchingAmounts || !lineItems) {
+    return null;
+  }
+  const relevantLineItems = lineItems.filter(
```
Comment (mbe1987) [inline paypal-commercetools-extension/src/utils/map.utils.ts:187]: I thought that you would actually keep the GiftLineItems but change the unit_amount to 0

**Review by mbe1987 — CHANGES_REQUESTED** [review]

File: paypal-commercetools-extension/src/utils/map.utils.ts (hunk 3 above)
Comment (LiudmylaMasliuk) [reply paypal-commercetools-extension/src/utils/map.utils.ts:187]: Tried - PayPal returns error if quantity is 0, also for prices there is a specification in the api that it should be >0, not >0

**Pull request merged**